app/routes/session.py

```diff
@@ -73,6 +73,32 @@ def login() -> Response:
 
     validate_jti(decrypted_token)
 
+    ru_ref = (
+        decrypted_token.get("survey_metadata", {}).get("data", {}).get("ru_ref")
+        if decrypted_token.get("survey_metadata")
+        else decrypted_token.get("ru_ref")
+    )
+
+    qid = (
+        decrypted_token.get("survey_metadata", {}).get("data", {}).get("qid")
+        if decrypted_token.get("survey_metadata")
+        else decrypted_token.get("qid")
+    )
```

Comment:
I suggest simplifying this to improve readability/reduce duplication. You can assign survey metadata for reuse.

Either a simple if else.
```
_data = (
    survey_metadata.get("data", {})
    if (survey_metadata := decrypted_token.get("survey_metadata"))
    else decrypted_token
)
ru_ref, qid = _data.get("ru_ref"), _data.get("qid")
```

Reply:
Updated 👍